src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -269,15 +269,21 @@ internal async Task<IEnumerable<IRpcWorkerChannel>> GetInitializedWorkerChannels
                     }
                 }
             }
-            IEnumerable<IRpcWorkerChannel> workerChannels = webhostChannels == null ? _jobHostLanguageWorkerChannelManager.GetChannels() : webhostChannels.Union(_jobHostLanguageWorkerChannelManager.GetChannels());
-            IEnumerable<IRpcWorkerChannel> initializedWorkers = workerChannels.Where(ch => ch.State == RpcWorkerChannelState.Initialized);
+            IEnumerable<IRpcWorkerChannel> initializedWorkers = GetInitializedAndInvocationBuffersInitializedChannels(webhostChannels, _jobHostLanguageWorkerChannelManager.GetChannels());
             if (initializedWorkers.Count() > _maxProcessCount)
             {
                 throw new InvalidOperationException($"Number of initialized language workers exceeded:{initializedWorkers.Count()} exceeded maxProcessCount: {_maxProcessCount}");
             }
             return initializedWorkers;
         }
 
+        internal static IEnumerable<IRpcWorkerChannel> GetInitializedAndInvocationBuffersInitializedChannels(IEnumerable<IRpcWorkerChannel> webhostChannels, IEnumerable<IRpcWorkerChannel> jobhostChannels)
+        {
+            IEnumerable<IRpcWorkerChannel> workerChannels = webhostChannels == null ? jobhostChannels : webhostChannels.Union(jobhostChannels);
+            IEnumerable<IRpcWorkerChannel> initializedWorkers = workerChannels.Where(ch => ch.State.HasFlag(RpcWorkerChannelState.Initialized) && ch.State.HasFlag(RpcWorkerChannelState.InvocationBuffersInitialized));
```

Comment:
Could we move this logic inside of RpcWorkerChannel itself? Or is there a reason we want to expose state?

Reply:
Yes. Hold on. Updating the PR now :)

Reply:
Haha sorry impatient :P thanks!!